corehq/apps/cleanup/management/commands/delete_es_docs_in_domain.py

```diff
@@ -0,0 +1,28 @@
+import sys
+
+from django.core.management import BaseCommand
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.es import AppES, CaseES, CaseSearchES, FormES, GroupES, UserES
+from corehq.apps.es.registry import registry_entry
+from corehq.apps.es.transient_util import doc_adapter_from_info
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+
+    def handle(self, domain, **options):
+        domain_obj = Domain.get_by_name(domain)
+        if domain_obj and not domain_obj.doc_type.endswith('-Deleted'):
+            print(f"{domain} has not been deleted. This command is intended for use on deleted domains only.")
+            sys.exit(1)
```

Comment:
nit: I think raising a [CommandError](https://docs.djangoproject.com/en/3.2/howto/custom-management-commands/#command-exceptions) is the "Django way" to do this, although it has caveats (what the Django docs call a _"nicely-printed error message"_ is subjective IMO).

Not a blocker, just an "FYI" in case there wasn't a specific reason you chose this way.

Reply:
Ah I've definitely heard that before and feels cleaner to me. Thanks for flagging 👍🏻